### PR TITLE
fix: grid column and grid row tests

### LINKED ISSUE #37720: Bug in CSS Grid: Use grid-column to Control Spacing
**Describe your problem and how to reproduce it:**
Correct solution lets you not pass test. 
The `grid-column: 2 / 4;` is called in the next challenge and even being refered to as well.

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/responsive-web-design/css-grid/use-grid-column-to-control-spacing

**Tell us about your browser and operating system:**
* Browser Name:  Google Chrome
* Browser Version:  78.0.3904.97 (Official Build) (64-bit)
* Operating System: macOS

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Screenshot 2019-11-08 at 22 58 59

### PR DESCRIPTION
**Pull request by moT01**
I slightly changed up the test text as well. And added a test in the grid row challenge to check that the property is used.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Closes #37720

### CODE REVIEW

File: curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md [hunk 1]
```diff
@@ -34,8 +34,8 @@ Make the item with the class <code>item5</code> consume the last two columns of
 tests:
   - text: <code>item5</code> class should have a <code>grid-column</code> property.
     testString: assert($('style').text().replace(/\s/g, '').match(/\.item5{.*grid-column:.*}/g));
-  - text: <code>item5</code> class should have a <code>grid-column</code> property which results in the <code>div</code> with the <code>item5</code> consuming the last two columns of the grid.
-    testString: assert(hasCorrectSpacing());
+  - text: <code>item5</code> class should have a <code>grid-column</code> property which results it consuming the last two columns of the grid.
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md]:
```suggestion
  - text: <code>item5</code> class should have a <code>grid-column</code> property which results in it consuming the last two columns of the grid.
```

File: curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md [hunk 2]
```diff
@@ -34,8 +34,8 @@ Make the item with the class <code>item5</code> consume the last two columns of
...
+    testString: const colStart = getComputedStyle($('.item5')[0]).gridColumnStart; const colEnd = getComputedStyle($('.item5')[0]).gridColumnEnd; const result = colStart.toString() + colEnd.toString(); assert(result === "24" || result === "2-1" || result === "2span 2" || result === "2span2" || result === "span 2-1" || result === "-12" || result === "span 2span 2" || result === "span 2auto" || result === "autospan 2");
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md]: To make the code easier to read and make it easier to add other possible results as users find them, why not use the following:
```suggestion
    testString: assert(doesGrid5ConsumeLastTwoGrids());
```
and use my other suggestion in the `Before Test` section

File: curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md [hunk 3]
```diff
@@ -83,24 +83,6 @@ tests:
...
-  const item5Width = $('.item5').width();
-  const diff = Math.abs(contTwoPlusThreePlusGapWidth - item5Width);
-  /* To avoid rounding errors the largest allowed diff is set at 0.01px */
-  return diff <= 0.01; 
-};
-</script>
-```
-
-</div> 
-
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md]:
````suggestion

### Before Test
<div id='html-setup'>

```html
<script>
function doesGrid5ConsumeLastTwoGrids() {
  const colStart = getComputedStyle($('.item5')[0]).gridColumnStart;
  const colEnd = getComputedStyle($('.item5')[0]).gridColumnEnd;
  const result = colStart.toString() + colEnd.toString();
  const possibleResults = [
    "24", "2-1", "2span 2","2span2", "span 2-1",
    "-12", "span 2span 2", "span 2auto","autospan 2"
  ];
  return possibleResults.some(val => result === val);
}
</script>
```

</div> 
````

**Review by RandellDawson — CHANGES_REQUESTED** [review]
@moT01 See my suggested changes.

File: curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.english.md [hunk 4]
```diff
@@ -21,8 +21,10 @@ Make the element with the <code>item5</code> class consume the last two rows.
 
 ```yml
 tests:
-  - text: <code>item5</code> class should have a <code>grid-row</code> property that has the value of <code>2 / 4</code>.
-    testString: assert(code.match(/.item5\s*?{[\s\S]*grid-row\s*?:\s*?2\s*?\/\s*?4\s*?;[\s\S]*}/gi));
+  - text: <code>item5</code> class should have a <code>grid-row</code> property.
+    testString: assert($('style').text().replace(/\s/g, '').match(/\.item5{.*grid-row:.*}/g));
+  - text: <code>item5</code> class should have a <code>grid-row</code> property which results in it consuming the last two rows of the grid.
+    testString: const rowStart = getComputedStyle($('.item5')[0]).gridRowStart; const rowEnd = getComputedStyle($('.item5')[0]).gridRowEnd; const result = rowStart.toString() + rowEnd.toString(); assert(result === "24" || result === "2-1" || result === "2span 2" || result === "2span2" || result === "span 2-1" || result === "-12" || result === "span 2span 2" || result === "span 2auto" || result === "autospan 2");
```
Comment (RandellDawson) [inline curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-row-to-control-spacing.english.md]: I would also do something similar here like I suggested in the other challenge.

File: curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md (hunk 2 above)
Comment (moT01) [reply curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md]: Is this really necessary?

Comment (RandellDawson) [reply curriculum/challenges/english/01-responsive-web-design/css-grid/use-grid-column-to-control-spacing.english.md]: Necessary?  No, but it makes it easier to read and change the code later.  All of those `||` are unnecessary in my opinion.

**Comment by moT01** [comment]
I made some changes along the lines of what you suggested @RandellDawson - I left out the `before test` section and made it all part of the `testString`, but split up the lines to make it more readable. I think this is about the most readable and succinct we will get.

**Review by RandellDawson — APPROVED** [review]

**Review by thecodingaviator — APPROVED** [review]

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**